### PR TITLE
GSLUX-767: Support for profile routing in v3

### PR DESCRIPTION
**Pull request by AlitaBernachot**
### JIRA issue

https://jira.camptocamp.com/browse/GSLUX-767

### Description

- add new component to be used in v3 for profile support in v3 Routing
- improve performances display of geomarker on mark (added `useSpatialIndex`, `declutter`, `updateWhileAnimating`, `updateWhileInteracting` ol props on layer anf feature)
- add new possibility to deactivate profile through a ref passed to the composable

### CODE REVIEW

File: src/composables/map/profile-position.composable.ts [hunk 1]
```diff
@@ -23,10 +32,14 @@ let overlay: PositionVectorLayer | undefined // Shared overlay between all profi
  *
  * NB.If edition mode is 'editLine', the marker is hidden because there is already a default blue marker
  * when modifying feature is active.
- * @param dataset
+ * @param dataset Profile data used to construct and handle the positioning (can be set afterwards by setting directly useProfilePosition().prfileData)
```
Comment (tkohr) [inline src/composables/map/profile-position.composable.ts:35]:
```suggestion
 * @param dataset Profile data used to construct and handle the positioning (can be set afterwards by setting directly useProfilePosition().profileData)
```

File: src/bundle/stores/profile-infos_v3.store.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    /**
+     * Activate or deactivate positioning for infos, deactivation is need for eg.
+     * when infos panel is closed but the features are still on the map,
+     * in this case, we need to deactivate the geomarker (as we don't see the profile anymore)
+     */
+    const activePositioning_v3 = ref(true)
+
+    function setProfileData(
+      map: Map,
+      feature: Feature & DrawnFeature,
```
[the remark this answers is not in the document]
Comment (AlitaBernachot) [reply src/bundle/stores/profile-infos_v3.store.ts:32]: Indeed Drawfeature is enough but actually this was to indicate that the v3 may pass a Feature (as it has no DrawFeature yet). This is not very important as this v3 will be removed anyway, removing `Feature &` as suggested.

File: src/bundle/stores/profile-routing_v3.store.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    /**
+     * Activate or deactivate positioning for routing, deactivation is need for eg.
+     * when routing panel is closed but the routing features are still on the map,
+     * in this case, we need to deactivate the geomarker (as we don't see the profile anymore)
+     */
+    const activePositioning_v3 = ref(true)
+
+    function setProfileData(
+      map: Map,
+      feature: Feature & DrawnFeature,
```
[the remark this answers is not in the document]
Comment (AlitaBernachot) [reply src/bundle/stores/profile-routing_v3.store.ts:32]: yes, done

File: src/composables/map/profile-position.composable.ts [hunk 4]
```diff
@@ -47,6 +61,19 @@ export default function useProfilePosition(dataset?: ProfileData) {
...
+    active => {
+      activePositioning.value = active ?? true
+    }
+  )
+
+  watch(activePositioning, active => {
+    if (!active) {
+      overlay?.removeGeoMarker()
+    }
+  })
```
[the remark this answers is not in the document]
Comment (AlitaBernachot) [reply src/composables/map/profile-position.composable.ts]: sure, done

File: src/bundle/stores/profile-infos_v3.store.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (AlitaBernachot) [reply src/bundle/stores/profile-infos_v3.store.ts]: @tkohr FYI I forgot to clean in the other two files before merging into `main`, this is now done in https://github.com/Geoportail-Luxembourg/luxembourg-geoportail/pull/177.

**Pull request merged**